pkg/datapath/linux/arp/ping_unsupported.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018-2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !linux
```

Comment:
Why do we need this dummy module? The arping is called only from `pkg/datapath/linux/node.go`, so AFAIK cilium-agent should compile on non-linux machines too.

Reply:
There are a few things which are os specific unfortunately. I.e. `unix.AF_PACKET` from the previous and the current implementations is available in Linux only. The original `arping` has a BSD specific file. Since the package is under the `linux` folder my assumption was that no BSD support is needed, but it won't be hard to get it added.

Reply:
Yeah, but if you omit the module, would cilium-agent still compile when you set e.g. `GOOS=darwin`?

Reply:
I may be failing to find instructions on how to check that, but `GOOS=darwin make build` stops with a cgo error in Linux. Makefile also uses incompatible parameters for `grep` in my MacOS. However, I would expect the compilation for non-Linux platform to fail without that file. Also, if you are concerned about its presence, a sibling package `pkg/datapath/linux/route` uses a similar approach in `route_darwin.go`.

Reply:
OK, I've just double-checked, and I see that the `pkg/datapath/linux` (consumer of arping) is being imported by packages which suppose to compile on darwin, so then it makes sense to keep this module.